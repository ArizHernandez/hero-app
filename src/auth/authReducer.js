import { types } from "../types/types";

/* State 
{
  name: 'Marco',
  isLoged: true/false
}
*/ 

export const authReducer = ( state = {}, action ) => {

  switch ( action.type ) {
    case types.login:
        return { 
          ...action.payload, 
          isLoged: true 
        }
    case types.logout:
      return {
        isLoged: false
      }
  
    default:
      return state;
  }
}