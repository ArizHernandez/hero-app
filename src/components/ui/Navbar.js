import React, { useContext } from 'react'
import { Link, NavLink, useHistory } from 'react-router-dom'
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

export const Navbar = () => {

  const { user: { name }, dispatch } = useContext(AuthContext);
  const history = useHistory();

  const handleLogout = () => {
    history.replace('/login');
    
    dispatch({ type: types.logout })
  }

  return (
    <nav className="navbar navbar-expand-md navbar-dark bg-dark">
      <div className="container-fluid">
        <Link className="navbar-brand" to="/">Asociaciones</Link>
        <button className="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
          <span className="navbar-toggler-icon"></span>
        </button>
        <div className="collapse navbar-collapse" id="navbarSupportedContent">
          <ul className="navbar-nav me-auto mb-2 mb-lg-0 w-100">
            <li className="nav-item">
              <NavLink
                exact
                className="nav-link"
                activeClassName="active" 
                to="/marvel" 
              >
                Marvel
              </NavLink>
            </li>
            <li className="nav-item">
              <NavLink
                exact 
                className="nav-link" 
                activeClassName="active"
                to="/dc"
              >
                DC
              </NavLink>
            </li>
            <li className="nav-item">
              <NavLink
                exact 
                className="nav-link" 
                activeClassName="active"
                to="/search"
              >
                Search
              </NavLink>
            </li>
            <li className="nav-item ms-md-auto d-flex">
              
              <span className="nav-link text-info">
                { name }
              </span>

              <button
                className="nav-link btn"
                onClick={ handleLogout }
              >
                Logout
              </button>
              
            </li>
          </ul>
        </div>
      </div>
    </nav>
  )
}