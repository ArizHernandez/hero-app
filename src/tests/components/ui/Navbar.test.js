import { mount } from 'enzyme';
import { MemoryRouter, Router } from 'react-router';

import { AuthContext } from '../../../auth/AuthContext';
import { Navbar } from '../../../components/ui/Navbar';
import { types } from '../../../types/types';

describe('Pruebas en <Navbar />', () => {

  const historyMock = {
    replace: jest.fn(),
    listen: jest.fn(),
    createHref: jest.fn(),
    location: {},
  }
  
  const context = {
    user: {
      name: 'Ariz',
      isLoged: true
    },
    dispatch: jest.fn()
  }

  const wrapper = mount( 
    <AuthContext.Provider value={context}>
      <MemoryRouter>
        <Router history={ historyMock }>
          <Navbar />
        </Router>
      </MemoryRouter>
    </AuthContext.Provider>
  )

  afterEach(() => {
    jest.clearAllMocks();
  })

  test('Debe de cargar correctamente', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.text-info').text().trim()).toBe(context.user.name);
  });
  
  test('Debe de llamar el logOut y usar history', () => {
    wrapper.find('button.nav-link').prop('onClick')();

    expect(context.dispatch).toBeCalledWith({ type: types.logout });
    expect(historyMock.replace).toBeCalledWith('/login');
  });
})