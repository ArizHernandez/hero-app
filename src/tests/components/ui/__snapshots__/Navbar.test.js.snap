// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Navbar /> Debe de cargar correctamente 1`] = `
<nav
  className="navbar navbar-expand-md navbar-dark bg-dark"
>
  <div
    className="container-fluid"
  >
    <a
      className="navbar-brand"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarSupportedContent"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav me-auto mb-2 mb-lg-0 w-100"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            onClick={[Function]}
          >
            Marvel
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            onClick={[Function]}
          >
            DC
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            onClick={[Function]}
          >
            Search
          </a>
        </li>
        <li
          className="nav-item ms-md-auto d-flex"
        >
          <span
            className="nav-link text-info"
          >
            Ariz
          </span>
          <button
            className="nav-link btn"
            onClick={[Function]}
          >
            Logout
          </button>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
